Yoast/javascript
# Rewrite keyword researchers to use stemmer instead of creating forms

## Summary

This PR can be summarized in the following changelog entry:

* [yoastseo] Changes the way word forms are recognized in the text of a post or page.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

These steps should be tested on English, German and Dutch texts. A regression test should be executed comparing the old version (e.g., current RC) and this version.
* Create a post, page, custom post, category, tag, custom taxonomy
* Add a text, title, meta description, slug, image with an alt tag
* Choose a focus keyphrase, fill it in. Use different forms of your keyphrase in the text, title, meta description, slug, image with an alt tag
* In `SEO analysis`, check that every assessment that looks at the use of keyphrase returns correct results. Use https://github.com/Yoast/javascript/wiki/Scoring-SEO-analysis in case of doubts. Note that this PR only changes the way keyphrase forms are recognized in the text, but does not fix existing bugs. Therefore, if the result of an assessment is different from your expectations, please check if the same holds for the old version (13.1).

Related keyphrase and synonym functionality was not altered in this PR, so there should be no regression on this side. In order to see, which assessments also take synonyms into consideration, please consult https://github.com/Yoast/javascript/wiki/Scoring-SEO-analysis.
The metadescription templates (SEO > content types) functionality was also not altered in this PR. Please smoke test it.

Additionally, please pay attention at whether performance has changed in re-analysis of a text since 13.1. You can, for example, introduce a change in the text that affects keyphrase analysis (e.g., add another use of keyphrase) and time how long it takes for the analysis to renew.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Keyphrase analysis

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-149
